src/odemis/driver/test/ephemeron_test.py

```diff
@@ -0,0 +1,253 @@
+        time.sleep(0.1)  # Gives a bit of time to be ready
+        self.sed.data.subscribe(self.receive_ebeam_data)
+
+        # Data should arrive approximately at the same time
+        self.ebeam_received.wait(expected_duration * 1.5)
+        self.ebic_received.wait(10)  # Gives a bit of margin to the EBIC to receive the data
+        assert self.ebeam_received.is_set() and self.ebic_received.is_set()
+
+        self.ebic_data.metadata[model.MD_PIXEL_SIZE] = self.ebeam_data.metadata[model.MD_PIXEL_SIZE]
+        self.ebic_data.metadata[model.MD_POS] = self.ebeam_data.metadata[model.MD_POS]
+        hdf5.export("test_ebeam.h5", [self.ebeam_data, self.ebic_data])
+        assert self.ebeam_data.shape == self.ebic_data.shape
+
+    def receive_ebic_data(self, df, d: model.DataArray):
+        logging.debug("Received EBIC data")
+        self.ebic_data = d
+        self.ebic_received.set()
+        df.unsubscribe(self.receive_ebic_data)
+
+    def receive_ebeam_data(self, df, d: model.DataArray):
+        logging.debug("Received e-beam data")
+        self.ebeam_data = d
+        self.ebeam_received.set()
+        df.unsubscribe(self.receive_ebeam_data)
+
+
+class TestMightyEBICDetector(unittest.TestCase):
+    """
+    Test case to test the functionality of the driver for an Ephemeron MightyEBIC detector.
+    """
+    @classmethod
+    def setUpClass(cls):
+        cls.ebic_det = ephemeron.MightyEBIC(**KWARGS_EBIC)
+        cls.acquired_data: Optional[model.DataArray] = None
+        cls.dwell_time_values = [cls.ebic_det.dwellTime.range[0], 6e-6, 10e-6, 11.9e-6, 12e-6, 50e-6, 1.995e-3, 1.996e-3]
+        cls.resolution_values = [(140, 100), (300, 420), (2100, 2000)]
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.ebic_det.terminate()
+
+    def test_dwell_time(self):
+        """
+        Test for setting the dwell time of the detector.
+        """
+        print(self.ebic_det.dwellTime.range)
+        for dt in self.dwell_time_values:
+            self.ebic_det.dwellTime.value = dt
+            self.assertLessEqual(self.ebic_det.dwellTime.value, dt)
+
+    def test_acquisition(self):
+        """
+        Test acquisitions with various dwell times and resolutions.
+        """
+        # TODO: for now, this assumes the hardware pixel trigger is either sent separately, or always
+        # active. (For the simulator, it simulates as if is always active)
+
+        for i, dt in enumerate(self.dwell_time_values):
+            # set the resolution of the region of acquisition to 140 x 100, force it non-squared
+            if dt < 1e-3:
+                res = (140, 500 + i)
+            else:
+                # Don't make it too long for the long dwell time
+                res = (100 + i, 100)
+            self.ebic_det.resolution.value = res
+            self.ebic_det.dwellTime.value = dt
+
+            act_dt = self.ebic_det.dwellTime.value
+            logging.debug("Expected duration for %s px @ %s s: %s s", res, act_dt,
```

Comment:
why not make an assert that the expected duration is close to the actual duration? What is the use of this log?

Reply:
The long acquisitions can take ~20s, so it's nice to know when looking at the test cases that something is happening.
I'll add a check for the acquisition time. It's a bit tricky with the real hardware, as it depends on some external factor, but there is a minimum duration that is easy to check.